packages/util-rxjs/src/coldObservableProvider.ts

```diff
@@ -58,16 +62,33 @@ export const coldObservableProvider = <T>({
           ).pipe(
             retryBackoff({
               ...retryBackoffConfig,
+              maxRetries: retryBackoffConfig.maxRetries ?? 10,
               shouldRetry: (error) => {
-                if (retryBackoffConfig.shouldRetry && !retryBackoffConfig.shouldRetry(error)) return false;
+                logger.error(error);
+
+                if (retryBackoffConfig.shouldRetry) {
+                  const shouldRetry = retryBackoffConfig.shouldRetry(error);
+                  logger.debug(`Should retry: ${shouldRetry}`);
+
+                  if (!shouldRetry) {
+                    return false;
+                  }
+                }
 
                 if (error instanceof InvalidStringError) {
                   onFatalError?.(error);
                   cancelOnFatalError$.next(true);
+                  return false;
```

Comment:
This is not be required (unless there's a bug).
When `cancelOnFatalError$` emits, it causes `internalCancel$` to emit, which will complete the observable.

Reply:
I think we can drop the `cancelOnFatalError$` observable. It will cancel anyway because we return false, which will cause the retry mechanism to throw.